### PR TITLE
Alignment and configuration

### PR DESCRIPTION
**Pull request by mjgiarlo**
* Rename `SynchronizedMinter` and align its code with the new `Minter::Base`-based minter structure
* Add accessor for default minter class to `Config`
* Use configured minter class as default in service
* Sets the default minter class to `Minter::Db` (rather than `Minter::File` -- **breaking change** from past versions)
* Update README with instructions for overriding the minter and using `Minter::File`
* Add `#read` and `#write` methods to minter classes
* Add new rake tasks that can migrate minter state between persistence options (+ documentation for same)
* Add test coverage for new minter `#read`/`#write` methods

### CODE REVIEW

**Comment by mjgiarlo** [comment]
Change README to clarify that switching minters does not automatically resume the old minter, and does not do any clean up. Recommend that users do not switch between minters once it's setup.

File: README.md [hunk 1]
```diff
@@ -124,7 +136,7 @@ For more information about the format of Noid patterns, see pages 8-10 of the [N
 
 ### Custom minters
 
-If you don't want your minter's state to be persisted, you may also pass in your own minter.  First write up a minter class that looks like the following:
+If you don't want your minter's state to be persisted, you may also write and configure your own minter.  First write up a minter class that looks like the following:
 
 ```ruby
 class MyMinter
```
Comment (no-reply) [inline README.md]: `class MyMinter < ActiveFedora::Noid::Minter::Base` ?

Comment (no-reply) [inline README.md]: Does the example code need to change, too?

**Comment by mjgiarlo** [comment]
@no-reply README updated.

**Comment by no-reply** [comment]
@mjgiarlo that looks good. Do you still mean to pull in the test refactor & docs, or should I push forward with this and submit a follow-up PR?

**Comment by atz** [comment]
I think it is problematic to have extracted locking to outside the transaction.  Locking for update now happens in multiple places, including on every `read`.  Previously only a transaction could block another transaction.  Now a `read` can.  

At the very least, some explanation would be appreciated why that is an improvement.  `read` and `write!` can now block each other: each calls `instance` which calls `MinterState.lock.find_by`, so they are **not working on the same object**, but each has a lock.

`read` and `write!` being public (and superficially documented) methods is sorta a problem.  I mean, the Base class says `read` returns a Hash representing the minter state.  Both it and the concrete class should be able to be more specific.  (E.G., are the keys symbolic, string or mixed? what's this magic JSON marshalling?)  Also, since it is public, a consumer might reasonably call `read` and be unaware that a locking operation is the result.  

By my interpretation, `read` does not need `SELECT ... FOR UPDATE` for an instance it converts to a Hash, then throws away.  Unlike before, no update ever comes for that object!  The point of having the transaction was to put the read and the `save!` on the same object.  This commit effectively undoes that.

Lastly, this was a lot of decomposable changes for one commit.  👎

**Pull request merged**